### PR TITLE
wicked: add POD for wickedbase

### PR DESCRIPTION
**Pull request by jlausuch**
- Related ticket: https://progress.opensuse.org/issues/42641

### CODE REVIEW

File: lib/wickedbase.pm [hunk 1]
```diff
@@ -109,6 +135,15 @@ sub get_ip {
     return $ip;
 }
 
+=head2 get_current_ip
+
+  get_current_ip($ifc [, ip_version => v4])
+
+Gets the IP of a given interface by C<ifc>.
+
+IP version can be specified by the parameter C<ip_version]> which is considered v4 by default.
```
Comment (cfconrad) [inline lib/wickedbase.pm]: The parameter C<ip_version> chould be one of the values 'v4' or 'v6'.

File: lib/wickedbase.pm [hunk 2]
```diff
@@ -109,6 +135,15 @@ sub get_ip {
     return $ip;
 }
 
+=head2 get_current_ip
+
+  get_current_ip($ifc [, ip_version => v4])
```
Comment (cfconrad) [inline lib/wickedbase.pm]: Adding single quotes around `'v4'` makes sure that we  expect a string.

File: lib/wickedbase.pm [hunk 3]
```diff
@@ -121,13 +156,31 @@ sub get_current_ip {
...
 sub get_from_data {
     my ($self, $source, $target, %args) = @_;
     $source .= check_var('IS_WICKED_REF', '1') ? 'ref' : 'sut' if $args{add_suffix};
     assert_script_run("wget --quiet " . data_url($source) . " -O $target");
     assert_script_run("chmod +x $target") if $args{executable};
 }
 
+=head2 ping_with_timeout
+
+  ping_with_timeout(timeout => $timeout, ip => $ip [, ip_version => [v4|v6]])
```
Comment (cfconrad) [inline lib/wickedbase.pm]: `ip_version => 'v4'`

**Comment by jlausuch** [comment]
@asmorodskyi @cfconrad

File: lib/wickedbase.pm [hunk 4]
```diff
@@ -20,6 +20,17 @@ use testapi qw(is_serial_terminal :DEFAULT);
 use serial_terminal;
 use Carp;
 
+=head2 wicked_command
+
+  wicked_command($action => [ifup|ifdown|ifreaload], $iface)
+
+Executes wicked command given the action on the corresponding interface.
+
+The mendatory parameter C<action> specifies the action [ifup|ifdown|ifreaload].
```
Comment (asmorodskyi) [inline lib/wickedbase.pm]:
```suggestion
The mandatory parameter C<action> specifies the action [ifup|ifdown|ifreaload].
```

File: lib/wickedbase.pm [hunk 5]
```diff
@@ -20,6 +20,17 @@ use testapi qw(is_serial_terminal :DEFAULT);
...
+The mendatory parameter C<iface> specifies the interface which action will be executed on.
```
Comment (asmorodskyi) [inline lib/wickedbase.pm]:
```suggestion
The mandatory parameter C<iface> specifies the interface which action will be executed on.
```

File: lib/wickedbase.pm [hunk 6]
```diff
@@ -198,6 +296,7 @@ sub setup_openvpn_client {
     assert_script_run("sed \'s/device/$device/\' -i $openvpn_client");
 }
 
+
```
Comment (asmorodskyi) [inline lib/wickedbase.pm]: why ?

File: lib/wickedbase.pm (hunk 4 above)
Comment (jlausuch) [reply lib/wickedbase.pm]: done

File: lib/wickedbase.pm (hunk 5 above)
Comment (jlausuch) [reply lib/wickedbase.pm]: done

File: lib/wickedbase.pm (hunk 6 above)
Comment (jlausuch) [reply lib/wickedbase.pm]: corrected

File: lib/wickedbase.pm [hunk 7]
```diff
@@ -260,6 +382,14 @@ sub do_mutex {
     }
 }
 
+=head2 done
+
+  done()
```
Comment (asmorodskyi) [inline lib/wickedbase.pm]: I think it is too much , documentation of this method in place where it's declared is good enough

File: lib/wickedbase.pm [hunk 8]
```diff
@@ -270,6 +400,15 @@ sub done {
     $self->SUPER::done();
 }
 
+=head2 pre_run_hook
+
+  pre_run_hook()
```
Comment (asmorodskyi) [inline lib/wickedbase.pm]: same here `pre_run_hook` should be documented once - in place of declaration

File: lib/wickedbase.pm [hunk 9]
```diff
@@ -210,6 +316,14 @@ sub before_scenario {
     record_info($title, $text);
 }
 
+=head2 get_test_result
+
+  get_test_result($type, $ip_version => v4)
```
Comment (cfconrad) [inline lib/wickedbase.pm]:
```suggestion
  get_test_result($type, $ip_version => 'v4')
```

**Comment by asmorodskyi** [comment]
LGTM , what about you @cfconrad ?

**Pull request merged**